Source/WebKit/Platform/IPC/ArgumentCoders.cpp

```diff
@@ -41,23 +41,23 @@ void ArgumentCoder<CString>::encode(Encoder& encoder, const CString& string)
 {
     // Special case the null string.
     if (string.isNull()) {
-        encoder << std::numeric_limits<uint32_t>::max();
+        encoder << std::numeric_limits<size_t>::max();
         return;
     }
 
-    encoder << static_cast<uint32_t>(string.length());
+    encoder << validateType<size_t>(string.length());
```

Comment:
`encode.template encode<size_t>(string.length());`

Which is more appropriate? I don't know..

Reply:
Idea was to have compile-time validation of what the expected type returned by the size accessor is. This approach does it through a templated pass-through, avoiding adding static asserts everywhere.

Along with compile-time validation, it also specifies clearly the type that's going into the encoder, giving clear indication of the type that's expected on the decoding side. Just extra assurance against any future changes that might unknowingly end up tweaking these parts.

Reply:
Yeah, that part is good. 
What I mean is that there's an option:
Either:
a) validate via `encoder << validateType<size_t>(...)`
Or:
b) enforce via `encode.template encode<size_t>(...)`
Or:
c) enforce via `encoder << static_cast<size_t>(..)`


So the question is why validate instead of enforce?

If the enforce would be missing, in that case the validate would be missing aswell, so there's no additional security guarantee ? 
Maybe the benefit of validate would be if there are warnings of redundant static_casts?
Or that we think that a developer will remove the redundant static cast?

Reply:
Validating, as done here, forces adjustment to the type that's returned by the size accessor, no unwitting conversion to anything else allowed. It's a bit clunky to work with, and as seen it doesn't really cause anything to be fixed at the moment.

Enforcing can, on integer types, cause problematic conversions. But at the moment I think that's not happening in any of these specializations, so I would just switch to that. b) seems nicest, and it allows to eventually implement validation in a more refined way.

Reply:
I thought `encoder.template encoder<T>()` exists, but it doesn't. Can be added later. Going with c).